fl4health/clients/model_merge_client.py

```diff
@@ -0,0 +1,292 @@
+        assert self.model is not None
+        return self.parameter_exchanger.push_parameters(self.model, config=config)
+
+    def set_parameters(self, parameters: NDArrays, config: Config) -> None:
+        """
+        Sets the local model parameters transferred from the server using a parameter exchanger
+            to coordinate how parameters are set.
+
+        For the ModelMergeClient, we assume that initially parameters are being set to the parameters
+            in the nn.Module returned by the user defined get_model method. Thus, set_parameters is
+            only called once after model merging has occured and before federated evaluation.
+
+        Args:
+            parameters (NDArrays): Parameters have information about model state to be added to the relevant client
+                model but may contain more information than that.
+            config (Config): The config is sent by the FL server to allow for customization in the function if desired.
+            fitting_round (bool): Boolean that indicates whether the current federated learning round is a
+                fitting round or an evaluation round.
+                This is used to help determine which parameter exchange should be used for pulling parameters.
+                A full parameter exchanger is only used if the current federated learning round is the very
+                first fitting round.
+        """
+        assert self.initialized
+        self.parameter_exchanger.pull_parameters(parameters, self.model)
+
+    def fit(self, parameters: NDArrays, config: Config) -> Tuple[NDArrays, int, Dict[str, Scalar]]:
+        """
+        Initializes client, validates local client model on local test data and returns parameters,
+            test dataset length and test metrics. Importantly, parameters from Server, which is empty,
+            is not used to initialized the client model.
+
+        Args:
+            parameters (NDArrays): The parameters of the model to be used in fit.
+            config (NDArrays): The config from the server.
+
+        Returns:
+            Tuple[NDArrays, int, Dict[str, Scalar]]: The parameters following the local training along with the
+            number of samples in the local training dataset and the computed metrics throughout the fit.
+
+        Raises:
+            ValueError: If local_steps or local_epochs is not specified in config.
+        """
+        assert not self.initialized
+        self.setup_client(config)
+        assert self.metrics_reporter is not None
+        self.metrics_reporter.add_to_metrics_at_round(
+            1,
+            data={"fit_start": datetime.datetime.now()},
+        )
+
+        val_metrics = self.validate()
+
+        self.metrics_reporter.add_to_metrics_at_round(
+            1,
+            data={
+                "fit_metrics": val_metrics,
+            },
+        )
+
+        return self.get_parameters(config), len(self.test_loader.dataset), val_metrics  # type: ignore
```

Comment:
We have to do this for the other clients as well, its just happening in the setup_client where we extract the sample counts from the datasets. Since technically not all pytorch dataset objects have a __len__ method but in practice they almost always do

Reply:
I will move the definition into the setup_client, because ignoring there makes it ambiguous what the issue is. Thanks for flagging

Reply:
Ah right...I remember. Thanks for the explanation.